Gems/ROS2/Code/Source/Lidar/LidarRaycaster.cpp

```diff
@@ -99,18 +98,20 @@ namespace ROS2
             request->m_direction = direction;
             request->m_distance = m_range;
             request->m_reportMultipleHits = false;
-            request->m_filterCallback = [ignoredLayerIndex = this->m_ignoredLayerIndex, ignoreLayer = this->m_ignoreLayer](
-                                            const AzPhysics::SimulatedBody* simBody, const Physics::Shape* shape)
+
+            request->m_filterCallback =
+                [ignoredLayerIndices = this->m_ignoredLayerIndices](const AzPhysics::SimulatedBody* simBody, const Physics::Shape* shape)
             {
-                if (ignoreLayer && (shape->GetCollisionLayer().GetIndex() == ignoredLayerIndex))
-                {
-                    return AzPhysics::SceneQuery::QueryHitType::None;
-                }
-                else
+                for (auto ignoredLayerIndex : ignoredLayerIndices)
```

Comment:
It is easier to store layers in an unordered set and check if the set contains his point collision layer